cmd/gapir/cc/main.cpp

```diff
@@ -144,7 +145,28 @@ void android_main(struct android_app* app) {
 
     // Note if you want to create a disk cache create it under:
     // app->activity->internalDataPath
-    listenConnections(std::move(conn), nullptr, nullptr, Connection::NO_TIMEOUT, &memoryManager, crashHandler);
+    std::thread listening_thread([&]() {
+      listenConnections(conn.get(), nullptr, nullptr, Connection::NO_TIMEOUT,
+                        &memoryManager, crashHandler);
+    });
+    while (true) {
```

Comment:
Shouldn't this loop exit if `listenConnections` returns?

Reply:
I think we should not. The infinite loop should only terminate when requested by the system. Actually an early return here does not destroy or ends the activity.

I've also tested with early return, the result is, if the activity was ever been started, any 'startservice' calls from adb shell will cause ANR.